Add note on S3 replication (#3507)
elastic/docs-content: deploy-manage/tools/snapshot-and-restore/s3-repository.md

Instances residing in a public subnet in an AWS VPC will connect to S3 via the VPC’s internet gateway and not be bandwidth limited by the VPC’s NAT instance.

## Replicating objects [repository-s3-replicating-objects]

AWS S3 supports [replication of objects](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html), both within a single region and across regions.

> Unfortunately this replication is not compatible with {{es}} snapshots.

Reviewer: The original proposal is going to place the 5 lines in the same paragraph, i'm going to propose to divide the content in 3 different paragraphs.

```suggestion
AWS S3 supports [replication of objects](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html), both within a single region and across regions. However, this replication mechanism is not compatible with {{es}} snapshots.

```

The objects that {{es}} writes to the repository refer to other objects which already exist in the repository, and {{es}} only deletes an object from the repository after it becomes unreferenced by all other objects.

> AWS S3 replication will apply operations to the replica repository in a different order from the order in which {{es}} applies them to the primary repository, and this may leave the replica repository in an invalid state, with objects referring to other objects that do not exist.

Reviewer: ```suggestion
Because AWS S3 replication can apply operations to the replica repository in a different order than {{es}} applies them to the primary repository, the replica repository can become invalid, with objects referring to others that no longer exist.

```

> To replicate a repository's contents elsewhere you must follow the [repository backup](snapshot-and-restore/self-managed.md#snapshots-repository-backup) process.

Author: ++ thanks
